R/profile_plot.R

```diff
@@ -130,24 +131,31 @@ profile_plot <- function(mydir, rep, para, profilesummary) {
 
   dev.off()
 
-  maxyr <- min(profilesummary$endyrs)
+  maxyr <- min(profilesummary$endyrs + 1)
   minyr <- max(profilesummary$startyrs)
-
+  
   est <- rep$parameters[rep$parameters$Label == para, "Value", 2]
   sb0_est <- rep$derived_quants[rep$derived_quants$Label == "SSB_Virgin", "Value"]
   sbf_est <- rep$derived_quants[rep$derived_quants$Label == paste0("SSB_", maxyr), "Value"]
   depl_est <- rep$derived_quants[rep$derived_quants$Label == paste0("Bratio_", maxyr), "Value"]
 
   x <- as.numeric(profilesummary$pars[profilesummary$pars$Label == para, n])
-  like <- as.numeric(profilesummary$likelihoods[profilesummary$likelihoods$Label == "TOTAL", n] - rep$likelihoods_used[1, 1])
+  # determine whether to include the prior likelihood component in the likelihood profile
```

Comment:
I think this change isn't needed, and it caused a problem for the petrale R0 profile resulting in a profile plot that went below 0. (There was no prior on R0, but the `Parm_Priors` component has non-zero values from priors on other parameters). 

The r4ss function already removes the "Total without prior" line for parameters without priors here:
https://github.com/r4ss/r4ss/blob/eca30726bc6dc5e03b84774bccf117a277d11371/R/SSplotProfile.R#L226-L230.

If you exclude the prior likelihood on a parameter which DOES have a prior in the estimated value, then all points in the profile should be shifted up or down equally relative to the MLE so the differences should be fine without further change.

Reply:
I have confirmed that this code works as expected across parameters with and without prior regardless of the setting in the starter file about whether to include the prior likelihood contribution.  I specifically did a profile over R0 with non-zero Parm_priors contributions and the minimum of the likelihood profile was 0.  The only profiles that went below 0 were ones where a parameter was fixed at a value that did not align with the best model file. 

Profile over R0 with priors on other parameters and the inclusion of the prior likelihood set in the starter file:

![parameter_panel_SR_LN(R0)](https://github.com/pfmc-assessments/nwfscDiag/assets/6172110/20618978-7a30-48b2-8f1f-b14dbffc373e)

Profile over h with the base model fixed at a value that does not align with the best fit to the data:
![parameter_panel_SR_BH_steep](https://github.com/pfmc-assessments/nwfscDiag/assets/6172110/e104d1b2-a364-41b2-88fe-f7a942c2d266)

Reply:
Thanks for looking into this further @chantelwetzel-noaa. I just spent a long time thinking about this and I don't know what's going on.

If the profilesummary total likelihood and the base model report file total likelihood aren't matching, and the difference is equal to the prior likelihoods, then subtracting that value to make them match seems like a good solution. But in that case, maybe the thing that gets subtracted should be just the prior likelihoods for the base model, not the prior like from each model which may differ across the profile. Otherwise, I think that change from min to max in the profile in the top left plot won't match the total likelihood line in the piner panel version this plot.

Regardless, I'm put off thinking about this further until another week.